python/tank/log.py

```diff
@@ -123,46 +217,66 @@ def initialize_base_file_logger(cls, log_name):
             # there is already a log handler.
             # terminate previous one
-            sgtk_root_logger.removeHandler(cls._std_file_handler)
-            cls._std_file_handler = None
+            log.debug(
+                "Tearing down existing log handler '%s' (%s)" % (previous_log_name, self._std_file_handler)
+            )
+            self._root_logger.removeHandler(self._std_file_handler)
+            self._std_file_handler = None
 
-        # set up logging root folder
-        log_folder = LocalFileStorageManager.get_global_root(LocalFileStorageManager.LOGGING)
-        filesystem.ensure_folder_exists(log_folder)
+        # store new log name
+        self._std_file_handler_log_name = log_name
 
-        # generate log path
-        log_file = os.path.join(
-            log_folder,
-            "%s.log" % filesystem.create_valid_filename(log_name)
-        )
+        if log_name:
+            # set up a new handler
 
-        cls._std_file_handler = logging.handlers.RotatingFileHandler(
-            log_file,
-            maxBytes=4096*1024,
-            backupCount=5
-        )
+            # avoid cyclic references
+            from .util import filesystem
 
-        # example:
-        # 2016-04-25 08:56:12,413 [44862 DEBUG tank.log] message message
-        formatter = logging.Formatter(
-            "%(asctime)s [%(process)d %(levelname)s %(name)s] %(message)s"
-        )
+            # set up logging root folder
+            filesystem.ensure_folder_exists(self.log_folder)
 
-        cls._std_file_handler.setFormatter(formatter)
-        sgtk_root_logger.addHandler(cls._std_file_handler)
+            # generate log path
+            log_file = os.path.join(
+                self.log_folder,
+                "%s.log" % filesystem.create_valid_filename(log_name)
+            )
 
-        # log the fact that we set up the log file :)
-        log.debug("Writing to log standard log file %s" % log_file)
 
+            # create a rotating log file with a max size of 18 megs -
+            # zendesk's current max attachment limit is 20mb so
+            # this should make all log files attachable to support tickets.
+            self._std_file_handler = logging.handlers.RotatingFileHandler(
+                log_file,
+                maxBytes=1024*1024*18,  # 18 MiB
+                backupCount=0
+            )
+
+            # example:
+            # 2016-04-25 08:56:12,413 [44862 DEBUG tank.log] message message
+            formatter = logging.Formatter(
+                "%(asctime)s [%(process)d %(levelname)s %(name)s] %(message)s"
+            )
+
+            self._std_file_handler.setFormatter(formatter)
+            self._root_logger.addHandler(self._std_file_handler)
+
+            # log the fact that we set up the log file :)
+            log.debug("Writing to log standard log file %s" % log_file)
+
+        # return previous log name
+        return previous_log_name
 
 def log_timing(name=None):
```

Comment:
Btw, if name is None, it would be nice if we could take the name of the method being decorated.